src/main/groovy/pl/allegro/tech/build/axion/release/domain/scm/ScmInitializationOptions.groovy

```diff
@@ -23,12 +23,25 @@ class ScmInitializationOptions {
         this.remoteUrl = remoteUrl
     }
 
+    static Map<String, ?> extractParameters(Project project) {
+        return [ATTACH_REMOTE_PROPERTY : extractOptionalParameter(project, ATTACH_REMOTE_PROPERTY),
+                FETCH_TAGS_PROPERTY : extractOptionalParameter(project, FETCH_TAGS_PROPERTY)]
+    }
+
+    private static def extractOptionalParameter(Project project, String paramName) {
+        return (String) (project.hasProperty(paramName) ? project.property(paramName) : null)
+    }
+
     static ScmInitializationOptions fromProject(Project project, String remote) {
+        return fromProject(extractParameters(project), remote)
```

Comment:
You can just use `fromProject(project.getProperties(), remote)`. There is no need to construct an intermediate map for holding these two options.